targets/TARGET_NUVOTON/TARGET_NUC472/trng_api.c

```diff
@@ -82,19 +87,20 @@ void trng_free(trng_t *obj)
 int trng_get_bytes(trng_t *obj, uint8_t *output, size_t length, size_t *output_length)
 {
     (void)obj;
-
+    unsigned char tmpBuff[32];
+        
     *output_length = 0;
-    if (length < 32) {
-        unsigned char tmpBuff[32];
+    
+    for (unsigned i = 0; i < (length/32); i++) {
+        trng_get(output);
+        *output_length += 32;
+        output += 32;
+    }     
+    if( length > *output_length ) {
         trng_get(tmpBuff);
-        memcpy(output, &tmpBuff, length);
+        memcpy(output, &tmpBuff, (length - *output_length));
```

Comment:
One cannot reason about the correctness of this because `*output_length` might have changed since the check `length > *output_length` -- it's not under our control. My suggestion would be to use an additional temporary to keep track of the number of bytes written so far, and to set `*output_length` to that temporary in the end.

Reply:
@hanno-arm  Agree with your opinion to avoid the risk of `*output_length's` change. 
Commit  2ee058b